### PR TITLE
add no right turn on red for traffic light

### PR DESCRIPTION
**Pull request by msbeta**
add no right turn on red for traffic light & support different sample points size between left and right boundary

### CODE REVIEW

File: modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h [hunk 1]
```diff
@@ -50,8 +50,18 @@ class LanesXmlParser {
   static Status ParseSpeed(const tinyxml2::XMLElement& xml_node, PbLane* lane);
   static Status ParseSampleAssociates(const tinyxml2::XMLElement& xml_node,
                                       PbLane* lane);
-  static Status ParseRoadSampleAssociates(const tinyxml2::XMLElement& xml_node,
-                                          PbLane* lane);
+  static Status ParseSingleSideRoadSampleAssociates(
+                                    const tinyxml2::XMLElement& xml_node,
+                                    PbLane* lane, bool bleft);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h]: Put input arguments ahead of output arguments.

File: modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h [hunk 2]
```diff
@@ -50,8 +50,18 @@ class LanesXmlParser {
...
+  static Status ParseLeftRoadSampleAssociates(
+                                    const tinyxml2::XMLElement& xml_node,
+                                    PbLane* lane);
```
Comment (xiaoxq) [inline modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h]: Try `scripts/clang-format.sh` to fix the style issues. Several indents are wrong in the pull request.

File: modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h (hunk 1 above)
Comment (msbeta) [reply modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h]: got，thanks.

File: modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h (hunk 2 above)
Comment (msbeta) [reply modules/map/hdmap/adapter/xml_parser/lanes_xml_parser.h]: I have fixed the style issues.

**Pull request merged**